tinker/static/assets/js/tinker.js

```diff
@@ -14,4 +14,117 @@ $(document).ready(function () {
             $(this).text("- Less Search Filters")
         }
     });
-});
\ No newline at end of file
+});
+
+function pagination() {
+    var numberOfItems = $(" #loop .list-group").length;
+    var limitPerPage = $("#first").val();
+
+    $("#selected-option").change(function() {
+        var o = $(this).find("option:selected");
+        if (o.val() === "All") {
+            limitPerPage = numberOfItems;
+        } else {
+            limitPerPage = o.val()
+        }
+
+        $(".pagination").children("li.to-remove").remove();
+
+        var totalPages = Math.ceil(numberOfItems / limitPerPage);
+
+        createButtons(totalPages, limitPerPage);
+
+        var currentPage = 1;
+
+        var grandTotal = limitPerPage * currentPage;
+
+        for (var i = grandTotal - limitPerPage; i < grandTotal ; i ++) {
+            $("#loop .list-group:eq(" + i + ")").show();
+        }
+
+        switchPages(limitPerPage);
+
+        nextPage(totalPages, limitPerPage);
+
+    });
+
+    var totalPages = Math.ceil(numberOfItems / limitPerPage);
+
+    createButtons(totalPages, limitPerPage);
+
+    switchPages(limitPerPage);
+
+    nextPage(totalPages, limitPerPage);
+
+    previousPage(totalPages, limitPerPage);
```

Comment:
are these 4 methods duplicated above?(45 and 47, etc)

Reply:
Only three methods are duplicated, and they are duplicated since they need to run again whenever we change the limit.

```CreateButtons``` is duplicated above since we need to re-append the buttons when we change the limit since we now (potentially) have a different number of pages.

```nextPage``` is duplicated above since we need to re-append it to the html after we change the limit.

Also ```switchPages``` is duplicated in the same way since some of the variables like grandTotal need to be updated when the limit is changed.

Reply:
I feel like those 3 can be avoided?

Reply:
👍